src/components/CIPipelineN/CreatePluginModal/CreatePluginFormContent.tsx

```diff
@@ -0,0 +1,394 @@
+    const { currentTab, name, pluginIdentifier, docLink, pluginVersion, description, tags, inputVariables, icon } =
+        pluginForm
+
+    const handleTabChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const targetMode = e.target.value as CreatePluginFormViewType
+
+        if (targetMode === currentTab) {
+            return
+        }
+
+        handleChange({ action: CreatePluginActionType.UPDATE_CURRENT_TAB, payload: targetMode })
+    }
+
+    const handlePluginSelection: SelectPickerProps['onChange'] = (newValue: SelectPickerOptionType) => {
+        handleChange({
+            action: CreatePluginActionType.UPDATE_PARENT_PLUGIN,
+            payload: { id: +newValue.value, name: newValue.label as string },
+        })
+    }
+
+    const renderPluginName = () => {
+        if (currentTab === CreatePluginFormViewType.EXISTING_PLUGIN) {
+            const { options, selectedOption } = getSelectPickerOptionsFromParentPluginList(parentPluginList, name)
+
+            return (
+                <SelectPicker
+                    inputId="select-existing-plugin"
+                    name="select-existing-plugin"
+                    classNamePrefix="select-existing-plugin"
+                    label="Existing Plugin"
+                    icon={<ICLegoBlock className="dc__no-shrink" />}
+                    required
+                    options={options}
+                    value={selectedOption}
+                    isLoading={isLoadingParentPluginList}
+                    optionListError={parentPluginListError}
+                    reloadOptionList={reloadParentPluginList}
+                    autoFocus
+                    placeholder="Select plugin"
+                    size={ComponentSizeType.large}
+                    // Should we disable this or allow user to select the plugin since we already have abortController?
+                    isDisabled={isLoadingParentPluginList || arePluginDetailsLoadingOrAborted}
+                    onKeyDown={stopPropagation}
+                    onChange={handlePluginSelection}
+                />
+            )
+        }
+
+        return (
+            <CreatePluginFormField
+                label="Plugin display name"
+                value={name}
+                error={pluginFormError.name}
+                action={CreatePluginActionType.UPDATE_NEW_PLUGIN_NAME}
+                handleChange={handleChange}
+                placeholder="Enter plugin name"
+                required
+                autoFocus
+            />
+        )
+    }
+
+    // Creating list of unique options from available tags and selected tags
+    const availableTagsOptions: OptionType[] =
+        availableTags?.map((tag) => ({
+            label: tag,
+            value: tag,
+        })) ?? []
+
+    const selectedTags: OptionType[] = tags.map((tag) => ({
+        label: tag,
+        value: tag,
+    }))
+
+    const tagOptions: OptionType[] = [...new Set([...availableTagsOptions, ...selectedTags])]
```

Comment:
please verify if set is working as intended

Reply:
Done replaced with set of string